### ISSUE TITLE
`no-return-array-push`: Ignore in `forEach`

### ISSUE THREAD

**Issue by boris-petrov** [sindresorhus/eslint-plugin-unicorn#3274]
### Description

`forEach` doesn't care about the return value so `no-return-array-push` shouldn't be triggered in it.

### Examples

```js
const x = [];
[].forEach(a => x.push(a));
```

**Comment by sindresorhus** [comment]
Thanks for reporting, but I'm going to decline this.

`a => x.push(a)` is a concise-body arrow whose expression evaluates to `push`'s return value (the new array length). Even though `forEach` discards it, writing it this way is exactly the smell the rule is meant to surface: it reads as if the return value is meaningful when it isn't. The fix is trivial and the rule points right at it:

```js
[].forEach(a => {
  x.push(a);
});
```

And really, the `forEach`-to-copy is itself the anti-pattern here. The whole thing collapses to:

```js
x.push(...someArray);
```

There's also a design reason. The rule is intentionally syntax-only and does not track types or call context. Ignoring `forEach` would mean inspecting what the arrow is passed to, and `forEach` isn't special: any callback that ignores its return value (a custom `each`, event handlers, etc.) has the same property. Carving out one built-in would be an arbitrary, incomplete half-measure that makes the rule less predictable.
